### DISCUSSION TITLE
500 Internal Server Error after upgrade

### DISCUSSION THREAD

**Discussion by VanessaMuraya** [kimai/kimai#5217] [Questions & Answers]
Hi,
I use Kimai 2.0.17 on Apache2 with PHP8.1 and MariaDB. Everythings runs well. But after an upgrade discribed here [https://www.kimai.org/documentation/updates.html] with "git checkout 2.1.0" I got an 500 error in my browser:
Oops! An Error Occurred
The server returned a "500 Internal Server Error".

Can somebody explain me what was happend wrong at this upgrade?

Thanks
Vanessa

**Comment by sebarreto** [comment]
Hi Vanessa, how are you?

Were you able to resolve this issue? I'm having the same problem. I have Kimai running on an Azure VM after upgrading to Ubuntu (22.04). I also upgraded PHP from 7.4 to 8.1. Currently, Kimai works, but only in the English version when the URL includes /en/, even though that's the only language that works. I've checked the locales, paths, services, translations... everything seems correct.

Thank you!

Best regards!

**Reply by VanessaMuraya** [reply]
I don't know it now anymore. I consulted the AI ​​back then. It was something like this:

> Hi! A 500 Internal Server Error after upgrading Kimai from 2.0.17 to 2.1.0 usually indicates something went wrong during the upgrade process. Based on the official upgrade documentation and common issues seen with Symfony-based apps like Kimai, here are the **likely causes and steps to fix it**:

---

## ✅ Common Causes of 500 Error After Kimai Upgrade

### 1. **Missing `composer install`**

After checking out the new version, you must run:

```bash
composer install --no-dev --optimize-autoloader
```

This updates the vendor libraries and autoloader. Without this, your code will be incompatible with the upgraded version and throw errors.

---

### 2. **Cache Not Cleared**

Symfony (which Kimai is built on) relies heavily on caching. After the upgrade, clear the cache:

```bash
bin/console cache:clear --env=prod
```

Also, make sure permissions are correct on the `var/` directory:

```bash
chmod -R 775 var/
chown -R www-data:www-data var/
```

Replace `www-data` with your actual web server user, if different.

---

### 3. **Database Not Updated**

You must update the database schema after the upgrade:

```bash
bin/console doctrine:migrations:migrate
```

This applies any new database changes Kimai 2.1.0 needs.

---

### 4. **PHP Version or Missing Extensions**

You’re using PHP 8.1, which should be fine for Kimai 2.1.0, but check that all required PHP extensions are enabled. Run:

```bash
php -m
```

Compare with Kimai's [[requirements](https://www.kimai.org/documentation/requirements.html)](https://www.kimai.org/documentation/requirements.html). Key extensions include `pdo_mysql`, `intl`, `xml`, `ctype`, `json`, etc.

---

### 5. **Apache Configuration or Permissions**

Check your Apache error log for detailed output:

```bash
tail -n 100 /var/log/apache2/error.log
```

You may find a specific error message pointing to the actual problem (e.g., missing file, permission denied, class not found).

---

## 🔧 Recommended Full Upgrade Steps

If you'd like to double-check your upgrade, here's a complete checklist:

```bash
# Make sure you're in the Kimai root directory
cd /path/to/kimai

# Update to the new version
git fetch --all
git checkout 2.1.0

# Install required dependencies
composer install --no-dev --optimize-autoloader

# Clear the cache
bin/console cache:clear --env=prod

# Set permissions
chown -R www-data:www-data var/
chmod -R 775 var/

# Update database schema
bin/console doctrine:migrations:migrate
```

---

## 📌 Tip

If the error persists, try enabling Symfony debug mode temporarily by setting:

```bash
APP_ENV=dev
APP_DEBUG=1
```

In your `.env` file or web server config. This will display detailed error messages in the browser instead of just "500", helping identify the exact issue.

---

Would you like help interpreting your Apache logs or Symfony debug output? If you paste the logs or enable debug mode and share the error message, I can pinpoint it further.

I looked at the suggestions and one of them worked. It's faster than with this Viennese maintainer. He had never answered my question, what for a guy!
